src/internal/components/focus-portal/index.tsx

```diff
@@ -0,0 +1,132 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useRef } from 'react';
+
+import Portal from '../portal';
+import FocusDetector from '../focus-detector';
+import { getFirstTabbable, getLastTabbable, getTabbables } from '../../utils/tabbables';
+
+export interface FocusPortalProps {
+  container?: Element;
+  children?: React.ReactNode;
+}
+
+export default function FocusPortal({ container, children }: FocusPortalProps) {
+  const proxyBeforeRef = useRef<HTMLDivElement | null>(null);
+  const proxyAfterRef = useRef<HTMLDivElement | null>(null);
+  const wrapperRef = useRef<HTMLDivElement | null>(null);
+  const portalBeforeRef = useRef<HTMLDivElement | null>(null);
+  const portalAfterRef = useRef<HTMLDivElement | null>(null);
+
+  const focusBeforeProxy = () => {
+    const allTabbableElements = getTabbables(document.documentElement);
```

Comment:
should you filter out all focus detectors rendered here?

Like `getTabbables(document.documentElement).filter(el => el !== proxyBeforeRef.current && el !== ...)`

Reply:
That is likely to fix the potential of an infinite loop mentioned below